rust-runtime/aws-smithy-types/src/retry.rs

```diff
@@ -230,6 +259,20 @@ pub struct RetryConfig {
     mode: RetryMode,
     max_attempts: u32,
     initial_backoff: Duration,
+    reconnect_mode: ReconnectMode,
+}
+
+/// Mode for connection re-establishment
+///
+/// By default, when a transient error is encountered, the connection in use will be poisoned. This
+/// behavior can be disabled by setting [`ReconnectMode::NoReconnect`] instead.
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub enum ReconnectMode {
+    /// Reconnect on [`ErrorKind::TransientError`]
+    ReconnectOnTransientError,
+
+    /// Disable reconnects
+    NoReconnect,
```

Comment:
The name here is probably good enough, but I could see how it could potentially cause confusion with reconnecting on hang up or something.

Reply:
Maybe `NoResuseBadConnection`? Usage is clearer from the tests / examples